### DISCUSSION TITLE
Auto-import of `/utils` function overrides component prop

### DISCUSSION THREAD

**Discussion by vincerubinetti** [nuxt/nuxt#20398] [Questions]
I have the following component:

```vue
<!-- 
  component to conditionally wrap slot content in tag
-->

<template>
  <component :is="tag" v-if="wrap">
    <slot />
  </component>
  <slot v-else />
</template>

<script setup lang="ts">
interface Props {
  /** tag */
  tag: string;
  /** wrap */
  wrap: boolean;
}

defineProps<Props>();
</script>
```

And I'm getting a typescript warning on `v-if="wrap"`:

`This condition will always return true since this function is always defined. Did you mean to call it instead?`

Because I have a function in `/utils/math.ts` called `wrap`. I'd expect the type from `props` to take precendence.

Is there a suggested way to handle this? Maybe this is a bug.

**Comment by peterroe** [comment]
This may indeed be a bug and need be fixed by nuxt team, by the way ,i advice you rename your function in utils and give a more longer and  more unique name to prevent hidden dangers in your project. 💚
